### PR TITLE
test(pager): add accessibility tests

### PR DESCRIPTION
**Pull request by divyajindel**
### Proposed behaviour
- Add `accessibility` Cypress tests for the `Pager` component to use the new cypress-component-react framework for testing.

### Current behaviour
Old approach for accessibility tests

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- [x] Run `npx cypress open --component` to check if there are newly added tests
- [x] Check if the `pager.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
- [x] Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that `Pager` tests are not running twice.

### CODE REVIEW

File: src/components/pager/pager-test.stories.tsx [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import React from "react";
 import { action } from "@storybook/addon-actions";
 import Pager, { PagerProps } from ".";
+import Form from "../form";
+import useMediaQuery from "../../hooks/useMediaQuery";
 
 export default {
   title: "Pager/Test",
+  includeStories: "Default",
```
Comment (DlgSHi) [inline src/components/pager/pager-test.stories.tsx:9]: where do you have a `Default` story here?

File: src/components/pager/pager.test.js [hunk 2]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
       currentPageWrapper().should("have.css", "margin-bottom", "0px");
     });
   });
+
+  describe("Accessibility tests for Pager component", () => {
+    it.each([[2], [5], [7]])(
```
Comment (DlgSHi) [inline src/components/pager/pager.test.js:489]:
```suggestion
    it.each([2, 5, 7])(
```

File: src/components/pager/pager.test.js [hunk 3]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+      "should render Pager with currentPage prop set to %s for accessibility tests",
+      (currentPage) => {
+        CypressMountWithProviders(<PagerComponent currentPage={currentPage} />);
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([[50], [100], [235]])(
```
Comment (DlgSHi) [inline src/components/pager/pager.test.js:497]:
```suggestion
    it.each([50, 100, 235])(
```

File: src/components/pager/pager.test.js [hunk 4]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+
+    it("both pager links are rendered when hideDisabledElements is set to true, but currenPage is above 1 for accessibility tests", () => {
+      CypressMountWithProviders(
+        <PagerComponent currentPage={7} hideDisabledElements />
+      );
+      cy.checkAccessibility();
+    });
+
+    describe("check funtionality for Pager component for accessibility tests", () => {
+      it.each([[-1], [-10], [-100], testData[0], testData[1]])(
```
Comment (DlgSHi) [inline src/components/pager/pager.test.js:639]:
```suggestion
      it.each([-1, -10, -100, testData[0], testData[1]])(
```

File: src/components/pager/pager.test.js [hunk 5]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+        (viewportWidth) => {
+          cy.viewport(viewportWidth, 768);
+
+          CypressMountWithProviders(<PagerComponentResponsive />);
+          cy.checkAccessibility();
+        }
+      );
+    });
+
+    describe("check events for Pager component", () => {
```
Comment (DlgSHi) [inline src/components/pager/pager.test.js:680]: events or accessibility?

File: src/components/pager/pager.test.js [hunk 6]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+      (bool) => {
+        CypressMountWithProviders(
+          <PagerComponent showPageSizeSelection={bool} />
+        );
+        cy.checkAccessibility();
+      }
+    );
+
+    // FE-4659
+    describe.skip("should render Pager component for accessibility tests", () => {
```
Comment (DlgSHi) [inline src/components/pager/pager.test.js:537]: why we are skipping all those tests? I can't see the accessibility violations on Storybook - but have the same config file

**Review by DlgSHi — CHANGES_REQUESTED** [review]

File: src/components/pager/pager.test.js [hunk 7]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+    );
+
+    it("both pager links are rendered when hideDisabledElements is set to true, but currenPage is above 1 for accessibility tests", () => {
+      CypressMountWithProviders(
+        <PagerComponent currentPage={7} hideDisabledElements />
+      );
+      cy.checkAccessibility();
+    });
+
+    describe("check funtionality for Pager component for accessibility tests", () => {
```
Comment (stephenogorman) [inline src/components/pager/pager.test.js:622]: Is there a reason the following tests are in a separate 'desribe'?

'funtionality' should be 'functionality'

File: src/components/pager/pager.test.js [hunk 8]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+          CypressMountWithProviders(
+            <PagerComponent totalRecords={totalRecords} />
+          );
+          cy.checkAccessibility();
+        }
+      );
+
+      it("should disable nextArrow and lastArrow buttons after clicking on lastArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={3} />);
+        cy.checkAccessibility();
```
Comment (stephenogorman) [inline src/components/pager/pager.test.js:635]: This test doesn't actually disable anything as nothing is clicked.

File: src/components/pager/pager.test.js [hunk 9]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+      });
+
+      it("should disable firstArrow and previousArrow buttons after clicking on firstArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={3} />);
+        cy.checkAccessibility();
```
Comment (stephenogorman) [inline src/components/pager/pager.test.js:640]: Same as previous comment.

File: src/components/pager/pager.test.js [hunk 10]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+      });
+
+      it("should disable firstArrow and previousArrow buttons after clicking on previousArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={2} />);
+        cy.checkAccessibility();
```
Comment (stephenogorman) [inline src/components/pager/pager.test.js:645]: Same as previous comment.

File: src/components/pager/pager.test.js [hunk 11]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+      });
+
+      it("should disable firstArrow and previousArrow buttons after clicking on nextArrow button for accessibility tests", () => {
+        CypressMountWithProviders(<PagerComponent currentPage={9} />);
+        cy.checkAccessibility();
```
Comment (stephenogorman) [inline src/components/pager/pager.test.js:650]: Same as previous comment.

File: src/components/pager/pager.test.js [hunk 12]
```diff
@@ -589,4 +484,266 @@ context("Test for Pager component", () => {
...
+        (viewportWidth) => {
+          cy.viewport(viewportWidth, 768);
+
+          CypressMountWithProviders(<PagerComponentResponsive />);
+          cy.checkAccessibility();
+        }
+      );
+    });
+
+    describe("check Pager events for accessibility", () => {
```
Comment (stephenogorman) [inline src/components/pager/pager.test.js:664]: I don't think these event tests need to be converted to accessibility tests. Nothing is actually being triggered in the test, so you're just testing the default component repeatedly.

File: src/components/pager/pager.test.js (hunk 8 above)
Comment (DlgSHi) [reply src/components/pager/pager.test.js:628]: it shouldn't :)

File: src/components/pager/pager.test.js (hunk 11 above)
Comment (DlgSHi) [reply src/components/pager/pager.test.js:665]: it shouldn't as well :)

File: src/components/pager/pager.test.js (hunk 12 above)
Comment (DlgSHi) [reply src/components/pager/pager.test.js:669]: +1

**Review by DlgSHi — DISMISSED** [review]

**Review by stephenogorman — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**